src/mbgl/renderer/tile_pyramid.cpp

```diff
@@ -198,13 +198,6 @@ void TilePyramid::update(const std::vector<Immutable<style::Layer::Impl>>& layer
         auto tilesIt = tiles.begin();
         auto retainIt = retain.begin();
         while (tilesIt != tiles.end()) {
-            auto renderedIt = rendered.find(tilesIt->first.toUnwrapped());
-            if (renderedIt == rendered.end()) {
-                // Since this tile isn't in the render set, crossTileIDs won't be kept
```

Comment:
I don't understand why this has become unnecessary, can you explain?

If a tile goes into the cache and gets removed from the `CrossTileSymbolIndex`, won't its `symbolInstances` still have their `crossTileID` set? Then when the bucket gets re-added, we re-run `findMatches` but it early exits because it sees there's already a `crossTileID` (which might no longer be correct)?

Reply:
Hmm, I guess the problematic case is this?

- tileA is loaded and label "Toronto" is assigned `crossTileID: 1`
- tileA is removed
- tileB is loaded and label "Toronto" is assigned `crossTileID: 2`
- tileA is loaded and label "Toronto" has the old id which doesn't match the id in tileBD

Reply:
Yup, that's what the original logic was added to catch. In practice, you can generate this scenario pretty easily by looking at a city at high zoom, panning to move the city out of the viewport, and then zooming out to bring the city back into the viewport.

Reply:
9df9ea233e06e4ca8a89365870956c6ae715f63e fixes the bug and adds a test that covers that case